saleor/order/calculations.py

```diff
@@ -122,6 +139,18 @@ def fetch_order_prices_if_expired(
         return order, lines
 
 
+def get_expired_line_ids(order: Order, lines: Iterable[OrderLine] | None) -> list[UUID]:
+    if order.status != OrderStatus.DRAFT:
+        return []
+
+    if lines is None:
```

Comment:
If you move this condition to `fetch_order_prices_if_expired`, then you could re-use already fetched lines in later part of the code:
```python
    if order.status not in ORDER_EDITABLE_STATUS:
        return order, lines

    if lines is None:
        lines = order.lines.all()

    expired_line_ids = get_expired_line_ids(order, lines)
```
Then rest of the calls will always have lines instead of None

Reply:
This way you make an extra db call for recalculations of orders with status `UNCONFIRMED`. That is why I moved it here. What do you think?